main.go

```diff
@@ -390,22 +392,42 @@ func SaveBackupAction(c *cli.Context) error {
 					"name":  backupName,
 					"error": err,
 				}).Warn("Error while trying to retrieve cluster state from cluster")
-
 			}
-			if err = CreateBackup(backupName, etcdCACert, etcdCert, etcdKey, etcdEndpoints, client, bc); err == nil {
+			compressedFilePath, err := CreateBackup(backupName, etcdCACert, etcdCert, etcdKey, etcdEndpoints, backupRetries)
+			if err == nil {
```

Comment:
Could you explain the logic in this block to me? I'm not sure why, if deleting a regular backup failed we are creating a new s3 backup and then deleting all s3 backups.. I may be missing something about how the functionality is supposed to work. Also, there should be a check after `DeleteS3Backups` that checks whether all backups have been removed like on line 399

Reply:
I've modified the logic to prevent unnecessary nesting, but it's still the same. It goes as follows:
- Try to create a backup
- If it failed, do nothing
- If it worked, rotate existing backups
- If we are supposed to be taking s3 backups, upload the backup
- If the upload worked, rotate existing s3 backups

Reply:
Oh I think the same point of confusion happened in another etcd snapshot PR.. where I didn't realize rotation meant deleting backups

Reply:
And there is where the backups are being rotated if the create was successful? https://github.com/rancher/rke-tools/blob/b8024f07b4b82d616c1fd0072cf2cdd4a684bed8/main.go#L363

Reply:
Cool. The

```
if err != nil
   continue
```
is kind of unintuitive.
Could you update this to `return err` is there's an error returned from a Create/Upload/Delete func, and if not then move through the code? Or let me know why it's done that way

Reply:
This section loops forever: it's for continuous backups so this is by design.

Reply:
Ok perf then